File: NewsDemo/app/src/test/java/com/kevin/newsdemo/data/JsonParseTest.java
package com.kevin.newsdemo.data;

import com.google.gson.Gson;
import org.junit.Test;

import static org.junit.Assert.assertEquals;

/**
 * Created by kevin on 2019/07/17 13:31.
 */
public class JsonParseTest {
    @Test
    public void test_parse_auth() {
        String json = "{\n" +
                "        \"id-token\": \"123456\",\n" +
                "        \"access-token\": \"98908989089\",\n" +
                "        \"refresh-token\": \"34545234234\"\n" +
                "    }";

        Gson gson = new Gson();
        Auth auth = gson.fromJson(json, Auth.class);
        assertEquals("123456", auth.getId());
        assertEquals("98908989089", auth.getToken());
        assertEquals("34545234234", auth.getRefreshToken());
    }

    @Test
    public void test_parse_user() {
        String json = "{\n" +
                "    \"auth\": {\n" +
                "        \"id-token\": \"123456\",\n" +
                "        \"access-token\": \"98908989089\",\n" +
                "        \"refresh-token\": \"34545234234\"\n" +
                "    }\n" +
                "}";

        Gson gson = new Gson();
        User user = gson.fromJson(json, User.class);
        Auth auth = user.getAuth();
        assertEquals("123456", auth.getId());
        assertEquals("98908989089", auth.getToken());
        assertEquals("34545234234", auth.getRefreshToken());
    }

    @Test
    public void test_parse_profile() {
        String json = "{\n" +
                "        \"name\": \"John Smith\",\n" +
                "        \"gender\": \"male\",\n" +
                "        \"avartar\": \"\"\n" +
                "    }";
        Gson gson = new Gson();
        Profile profile = gson.fromJson(json, Profile.class);
        assertEquals("John Smith", profile.getName());
        assertEquals("male", profile.getGender());
        assertEquals("", profile.getAvartar());
    }

    @Test
    public void test_parse_user_profile() {
        String json = "{\n" +
                "    \"profile\": {\n" +
                "        \"name\": \"John Smith\",\n" +
                "        \"gender\": \"male\",\n" +
                "        \"avartar\": \"\"\n" +
                "    }\n" +
                "}";
        Gson gson = new Gson();
        UserProfile userProfile = gson.fromJson(json, UserProfile.class);
        Profile profile = userProfile.getProfile();
        assertEquals("John Smith", profile.getName());
        assertEquals("male", profile.getGender());
        assertEquals("", profile.getAvartar());
    }
}

File: NewsDemo/app/src/test/java/com/kevin/newsdemo/data/JsonParseTest.kt
package com.kevin.newsdemo.data

import com.google.gson.Gson
import org.junit.Assert.assertEquals
import org.junit.Test

/**
 * Created by kevin on 2019/07/17 13:31.
 */
class JsonParseTest {
    @Test
    fun test_parse_auth() {
        val json = "{\n" +
                "        \"id-token\": \"123456\",\n" +
                "        \"access-token\": \"98908989089\",\n" +
                "        \"refresh-token\": \"34545234234\"\n" +
                "    }"

        val gson = Gson()
        val auth = gson.fromJson(json, Auth::class.java)
        assertEquals("123456", auth.idToken)
        assertEquals("98908989089", auth.token)
        assertEquals("34545234234", auth.refreshToken)
    }

    @Test
    fun test_parse_user() {
        val json = "{\n" +
                "    \"auth\": {\n" +
                "        \"id-token\": \"123456\",\n" +
                "        \"access-token\": \"98908989089\",\n" +
                "        \"refresh-token\": \"34545234234\"\n" +
                "    }\n" +
                "}"

        val gson = Gson()
        val user = gson.fromJson(json, User::class.java)
        val auth = user.auth
        assertEquals("123456", auth.idToken)
        assertEquals("98908989089", auth.token)
        assertEquals("34545234234", auth.refreshToken)
    }

    @Test
    fun test_parse_profile() {
        val json = "{\n" +
                "        \"name\": \"John Smith\",\n" +
                "        \"gender\": \"male\",\n" +
                "        \"avartar\": \"\"\n" +
                "    }"
        val gson = Gson()
        val (name, gender, avartar) = gson.fromJson(json, Profile::class.java)
        assertEquals("John Smith", name)
        assertEquals("male", gender)
        assertEquals("", avartar)
    }

    @Test
    fun test_parse_user_profile() {
        val json = "{\n" +
                "    \"profile\": {\n" +
                "        \"name\": \"John Smith\",\n" +
                "        \"gender\": \"male\",\n" +
                "        \"avartar\": \"\"\n" +
                "    }\n" +
                "}"
        val gson = Gson()
        val userProfile = gson.fromJson(json, UserProfile::class.java)
        val profile = userProfile.profile
        assertEquals("John Smith", profile!!.name)
        assertEquals("male", profile.gender)
        assertEquals("", profile.avartar)
    }
}
